lib/api_resources/peer_management.js

```diff
@@ -65,47 +72,166 @@ PeerManagementResource.prototype.link = function(env, next) {
+      } else {
+        env.response.statusCode = 500;
+        next(env);  
+      }
+    } else {
+      env.response.statusCode = 404;
+      next(env);
+    }
+  });
+};
+
+//Updating a peer is a two part process.
+//First we'll determine whether or not the API call should be proxied
+//Then we'll connect to the new peer.
+//Then we'll disconnect from the old peer.
+PeerManagementResource.prototype.updatePeer = function(env, next) {
+  var self = this;
+  env.request.getBody(function(err, body) {
+    var params = querystring.parse(body.toString());
+    var id = params.connectionId;
+    var url = params.url;
+    var query = Query.of('peers').where({connectionId: id});
+    self.registry.find(query, function(err, results) {
+      if(results.length) {
+        var peer = results[0];
+        if(peer.direction === 'initiator') {
+
+          self.server._peers.push(url);
+
+          var newPeer = {
+            url: url,
+            direction: 'initiator',
+            status: 'connecting'
+          }; 
+          
+
+          self.registry.add(newPeer, function(err, newPeer) {
+            if (err) {
+              env.response.statusCode = 500;
+              return next(env);
+            }
+
+            self._connect(newPeer);
+            self._disconnect(peer);
+            env.response.statusCode = 200;
+            next(env);
+          });
+
+        } else if(peer.direction === 'acceptor') {
+          self._proxyDisconnect(env, next, id);
+        } else {
+          env.response.statusCode = 500;
+          next(env);
+        } 
+      } else {
+        env.response.statusCode = 404;
+        next(env);
+      }
+    });
+  });
+};
+
+PeerManagementResource.prototype._disconnect = function(peer) {
+  var wsUrl = PeerClient.calculatePeerUrl(peer.url, this.server._name);
+  var peers = this.server._peerClients.filter(function(peer) {
+    return peer.url === wsUrl;  
+  });
+  var client = peers[0];
+  client.close();     
+};
+
+PeerManagementResource.prototype._proxyDisconnect = function(env, next, id) { 
+  var self = this;
+  var peerSocket;
+  var sockets = Object.keys(this.server.httpServer.peers).forEach(function(socketId){
+    var peers = self.server.httpServer.peers;
+    var socket = peers[socketId];
+    if(socket.connectionId === id) {
+      peerSocket = socket;  
+    }
+  });
+
+  peerSocket.on('end', function() {
+    env.response.statusCode = 200;
+    next(env);  
+  });
+
+  this._proxyToPeer(peerSocket, env);
+ 
+};
+
+PeerManagementResource.prototype._proxyToPeer = function(peer, env) {
+  var req = env.request;
+  var res = env.response;
+  var agent = env.zettaAgent || peer.agent;
+  var opts = { method: req.method, headers: req.headers, path: req.url, agent: agent };
+  var request = http.request(opts);
+  if(req.body) {
+    request.end(req.body);  
+  } else {
+    req.pipe(request);
+  }
+};
+
 PeerManagementResource.prototype._connect = function(peer) {
```

Comment:
This functionality isn't for linking. It's for proxying peering commands from an acceptor peer to an initiator peer. This uses a fire and forget request because we can't reliably listen for a response back up the pipe.